### ISSUE TITLE
`Style/HashConversion` cop is unsafe for arguments that are not enclosed in parentheses.

### ISSUE THREAD

**Issue by 1grc** [rubocop/rubocop#9672]
## Expected behavior

Safe auto correct of `Style/HashConversion` should not introduce errors in the code.

## Actual behavior
If the auto-correct target is an argument without parentheses, the line will be interpreted as a SyntaxError or `One-line pattern matching` after correct.

## Steps to reproduce the problem

Create `example.rb` :

```ruby
# frozen_string_literal: true

puts Hash['a', 'b']
```

Next, run the following to reproduce the error:  
```
bundle exec rubocop --debug --enable-pending-cops -a example.rb
```
```
$ bundle exec rubocop --debug --enable-pending-cops -a example.rb
For /home/kazuma/repo/test: Default configuration from /home/kazuma/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/rubocop-1.12.0/config/default.yml
Inspecting 1 file
Scanning /home/kazuma/repo/test/example.rb
E

Offenses:

example.rb:3:6: C: [Corrected] Style/HashConversion: Prefer literal hash to Hash[arg1, arg2, ...].
puts Hash['a', 'b']
     ^^^^^^^^^^^^^^
example.rb:3:11: E: Lint/Syntax: unexpected token tASSOC
(Using Ruby 2.4 parser; configure using TargetRubyVersion parameter, under AllCops)
puts {'a' => 'b'}
          ^^

1 file inspected, 2 offenses detected, 1 offense corrected
Finished in 0.14253549999557436 seconds
```
```ruby
# example.rb

# frozen_string_literal: true
  
puts {'a' => 'b'}
```

```
$ ruby example.rb 
example.rb:3: warning: One-line pattern matching is experimental, and the behavior may change in future versions of Ruby!
```

## RuboCop version

```
$ bundle exec rubocop -V
1.12.0 (using Parser 3.0.0.0, rubocop-ast 1.4.1, running on ruby 3.0.0 x86_64-linux)
```
